### DISCUSSION TITLE
MLIR GPU CodeGen for XLA

### DISCUSSION THREAD

**Discussion by kevint324** [openxla/xla#16] [General]
hi folks,

According to [MLIR CodeGen for XLA](https://www.tensorflow.org/mlir/xla_gpu_codegen),  most of the current XLA's codegen will move to  MLIR based implementation.

This page hasn't been updated for quite sometime. 
According to my understanding the current status is at "Migrating Device LLVM IR (Task 3)". Am I right?
`ElementalIrEmitter ops go for (4)` which is `Refactor old emitters to be LHLO -> Linalg, and write new Linalg emitters`
Is this still valid? Is this task on schedule?

Thanks

**Comment by joker-eph** [comment]
We're still moving the MLIR, but the plan has changed a little bit in terms of what the end-result should look like. We aim to update the end-to-end documentation by the end of next month.

**Reply by kevint324** [reply]
Cool. Thanks for the update.

**Reply by kevint324** [reply]
https://www.tensorflow.org/mlir/xla_gpu_codegen

This link is 404'ed. Is there a updated doc?
Thanks
